### PR TITLE
feat: allow specifying extra labels on the coderd deployment

### PR DESCRIPTION
**Pull request by johnstcn**
This commit allows users to specify arbitrary additional labels on pods in the coderd deployment.

### CODE REVIEW

**Review by f0ssel — APPROVED** [review]
Good addition

File: templates/coderd.yaml [hunk 1]
```diff
@@ -27,6 +27,9 @@ spec:
         app.kubernetes.io/part-of: coder
         app: {{ include "coder.serviceName" . }}
         coder.deployment: {{ include "coder.serviceName" . }}
+        {{- with .Values.coderd.extraLabels -}}
```
Comment (jawnsy) [inline templates/coderd.yaml:30]: I don't see any unit tests for this :stuck_out_tongue:

Comment (johnstcn) [reply templates/coderd.yaml:30]: Your unit tests caught the lack of unit tests! 🎉

File: values.yaml [hunk 2]
```diff
@@ -151,6 +151,9 @@ coderd:
       # contains the super admin password.
       key: "password"
 
+  # coderd.extraLabels -- Allows specifying additional labels to pods in the
+  # `coderd` deployment (.spec.template.metadata.labels).
+  extraLabels: {}
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply values.yaml:156]: `extraLabels` is easier to write? 🤷‍♂️

[the remark this answers is not in the document]
Comment (jawnsy) [reply values.yaml:156]: The bitnami charts are considered pretty good: https://github.com/bitnami/charts/tree/master/bitnami/postgresql

That one uses extraEnv, extraEnvVarsCM, etc

I don't have a strong opinion between extra/additional myself, as long as we're consistent in the way we do things.

**Pull request merged**